### DISCUSSION TITLE
Pre release version 3.7.0 release candidate 1

### DISCUSSION THREAD

**Discussion by ann0see** [jamulussoftware/jamulus#1217] [Announcements]
**THIS IS A PRE RELEASE!**

Softins just released the Release candidate 1 of Jamulus 3.7.0 and we would be happy if you downloaded and tested it for serious bugs (server crashes etc.) before 3.7.0 gets released finally.

Get the latest release candidate here: https://github.com/jamulussoftware/jamulus/releases/tag/r3_7_0rc1

ChangeLog (until now):

- introduced new server lists: Any Genre 3, Choral/Barbershop (#875),
  also renamed Default to Any Genre 1, Any Genre to Any Genre 2,
  and Classical/Folk/Choral to Classical/Folk.

- central server addresses are now in jamulus.io instead of fischvolk.de (#919)

- show --clientname as first word in title to avoid clipping in Windows task bar (#789)

- improved Windows installer, coded by ann0see and doloopuntil (#792, #841)

- support http links in the chat dialog, coded by hoffie (#879)

- support Github actions for CodeQL, Autobuild, Publish Release,
  coded by pljones, nefarius2001, ann0see, ranfdev.

- more detailed version number of unreleased version, coded by nefarius2001 (#475)

- add new --serverpublicip option to support central servers behind NAT,
  coded by hoffie (#954)

- replaced low-quality background image (fancy skin) of mixerboard, created by passing (#970)

- Escape chat messages at the server. This will no longer allow html formated chat messages due to security reasons,
  coded by atsampson (#939)

- Highlight the fact that the server is recording in the Mixerboard title (#968), coded by dcorson-ticino.com

- Implement new --ctrlmidich syntax allowing to specify fader, pan, mute,
  solo buttons (#945).  Implementation for mute&solo buttons for now is
  only for toggle controllers and does not support headless operation yet, coded by dakhubgit
  
- Move ASIO Setup button on Windows below driver selection (#977), coded by mulyaj and ann0see

- Gray out input levels and show a message if you're not connected to a server, coded by ann0see (#847, #983)

- bug fix: remove ConsoleWriterFactory - to resolve suspected memory leak, coded by pljones (#926)

- bug fix: fix crash if settings are changed in ASIO4ALL during a connection (contained in #796). Reverts #727 for Windows

- Avoid screen from sleeping or starting screen saver for Mac, coded by jerogee (#834)

- stop sending traffic to CloudFlare as part of the external IP detection
  logic (#633, #1092), suggested by atsampson, coded by hoffie

- bug fix: properly restore mixer levels after reconnecting to a restarted
  server. This is a server-side fix. coded by hoffie (#955, #1010)

- Improve update version detection, coded by softins (#1155)
  Check two servers instead of one (updatecheck1.jamulus.io and updatecheck2.jamulus.io).
  Ignore the server version if it is not a release, due to a suffix such as dev, alpha or beta.

- Windows installer interface now supports English, Dutch, French, German, Italian, Polish, Portuguese, Spanish and Swedish.

- bug fix: Fixed some memory leaks in server recording, coded by softins (#1073)

- bug fix: Adjusted some accelerator keys in translations to avoid duplication (#1165)

- Release now builds .deb installation packages for Ubuntu/Debian, coded by ann0see (#1100)
  Compatible with Ubuntu 18.04 or later.

- Release now builds experimental .apk installation packages for Android, coded by nefarius2001 (#880)

- bug fix: Added missing driverInfo initialisation in Windows sound driver, coded by npostavs (#1194)

**Comment by ann0see** [comment]
Windows macOS and Linux clients send and receive sound. 

Linux client has to be investigated further. There might be glitches if you switch between normal and fancy mode. Windows is a bit slow (but this could also be my impression only).

**Reply by hoffie** [reply]
> Linux client has to be investigated further. There might be glitches if you switch between normal and fancy mode. Windows is a bit slow (but this could also be my impression only).

So are you suspecting some regression? Been using the master code (~beta3) both on my servers and on my Linux clients for some time now.

**Reply by ann0see** [reply]
Probably. But it might also be a glitch based on something odd with my PC or GH Actions.

**Reply by ann0see** [reply]
I'll compile it on my linux machine now and check if the same bug comes up there.

**Reply by ann0see** [reply]
Ok. See: https://github.com/jamulussoftware/jamulus/discussions/1217#discussioncomment-464967

It seems to be a regression

**Comment by dcorson-ticino-com** [comment]
Saw some English "Yes" on buttons for the German installer.

**Reply by hoffie** [reply]
> Saw some English "Yes" on buttons for the German installer.

Thanks for reporting.

I can verify this when running the Windows installer under Linux in wine with `LC_ALL=C` and selecting `German` in the UI (it's the ASIO-not-found popup).
I cannot verify this when running the Windows installer under Linux in wine with `LC_ALL=de_DE` and selecting `German` in the UI.

I suspect that these strings come from NSI-native translations (or even from the OS?) as they are not in our language files.

Can you share details about your environment? Were you trying on Windows or with wine (like me)? What Windows version? Is your Windows language German or English?

**Reply by ann0see** [reply]
These buttons should be translated automatically. I doubt, we can do anything here.

**Comment by Zyfuss** [comment]
I found a little mistake in one of the installation popup (Italian). The correct word is "versione" (not versine)
![image](https://user-images.githubusercontent.com/80408727/110649941-61d23b80-81ba-11eb-80e6-e5e13beb5e2f.png)
Thanks. Ciao.

**Reply by wschell** [reply]
I have made a public testserver with 3.7.0rc1 in Düsseldorf, Genre Rock, named Testserver 3.7.0rc1.

**Reply by hoffie** [reply]
> I found a little mistake in one of the installation popup (Italian). The correct word is "versione" (not versine)

Thanks for reporting! :)
Will fix in #1221.

**Reply by ann0see** [reply]
I will mark this as resolved now.

**Comment by ann0see** [comment]
@wschell 

> I have made a public testserver with 3.7.0rc1 in Düsseldorf, Genre Rock, named Testserver 3.7.0rc1.

Which OS are you running it on? Headless Linux/Windows?

**Reply by wschell** [reply]
Ubuntu 20.04.2 LTS headless

**Reply by ann0see** [reply]
Great. Thank you for the information. So you used the official .deb file?

**Reply by wschell** [reply]
No, sorry - I made a "git pull" and make...
But I could install the package if this is useful for you. It would be later in the evening.

**Reply by ann0see** [reply]
Ok. I'd prefer the .deb. Note that it installs to a different location.

**Reply by wschell** [reply]
Now the headless.deb package is installed and both server are running and registered in genre Rock;

**Reply by ann0see** [reply]
Great! Thank you very much! I hope everything works fine with them.

**Comment by ann0see** [comment]
Seems as if I can trace down a bug to https://github.com/jamulussoftware/jamulus/commit/f5a58a0d2db7f38d8f6cdc5243b276062c9b2b8b

@dcorson-ticino-com

On Linux: Debian 10 Gnome after this commit the mixing board seems not to change back to fancy, normal, compact mode if I switch it during runtime. Will do some further investigation on this.

**Reply by dcorson-ticino-com** [reply]
How exactly did you test?  As far as I can see it is working OK under Win10 also with runtime changes of the skin.
D

**Reply by hoffie** [reply]
> How exactly did you test? As far as I can see it is working OK under Win10 also with runtime changes of the skin.

Switching skins seems to work fine on Arch Linux as well.

**Reply by ann0see** [reply]
1. Start Jamulus with normal skin enabled. 
2. Open settings and switch to fancy skin
3. Look at the mixer board: For me on Gnome Debian 10 it is not reset and remains white.

![bug](https://user-images.githubusercontent.com/20726856/110692483-a70c6280-81e6-11eb-9560-bc59f4ab43f8.png)

I have a line in suspect: 1412 in clientdlg.cpp in CClientDlg::SetMixerBoardDeco Not sure, but I think  eNewDesign == eLastDesign might be true in my case at least at the time of the change. I'll comment these lines and check again. Not sure *why* this should happen.

**Reply by hoffie** [reply]
Still works for me as expected, even when following the steps exactly.

**Reply by ann0see** [reply]
Maybe I mis-remembered: 
Now I could reproduce it like that:

1. Start Jamulus with fancy mode enabled
2. Switch to normal mode while settings is open
3. Switch back to fancy mode (maybe you'll need to switch back and forth between the different modes).

Note: I'm not connected to a server
I commented the lines I mentioned but it doesn't seem to be related to them.

**Reply by hoffie** [reply]
I just tried on Fedora 33 with Gnome and still cannot reproduce. :(
I tried both when being connected and when not.

> Switch to normal mode while settings is open

Just to be sure we are talking about the same: You mean the settings window where one switches the skin, right? So this window has to be open as there is no other way to switch, right?

You say it's a regression. Is there a Jamulus version which does not exhibit this problem for you?

I suspect that it's either some kind of rendering glitch (maybe depending on graphics card drivers?) or a bug which I fail to trigger... :)

**Reply by dcorson-ticino-com** [reply]
Under Win10 I can change skins as often as I want and all is OK.  
That picture shows that the Mixerboard is not rendered/painted at all.  
Very strange.  
The changes I did just involve the style, not the rendering/painting itself.
For the moment I don't really know where to look.

**Reply by ann0see** [reply]
> You mean the settings window where one switches the skin, right? So this window has to be open as there is no other way to switch, right?

Yes

> You say it's a regression. Is there a Jamulus version which does not exhibit this problem for you?

Yes. 3.6.2 didn't seem to have this bug. But I can re test it. 

I can't reproduce this bug on windows either.

**Reply by dcorson-ticino-com** [reply]
How do we go forward here?  Unfortunately I don't have a Linux machine to test on.

**Reply by ann0see** [reply]
That's a good question. Since I'm the only one with this issue, maybe it's related to something else? Dakhubgit also had some strange UI glitches.

**Reply by ghost** [reply]
@ann0see If you can compile Jamulus, you should bisect Jamulus until you find the problem. I have not used bisect, but you can manually build and test Jamulus until you find the offending commit.

**Reply by pljones** [reply]
I'm running a remote X session to my Linux box (so my "Window manager" is Windows 10) and couldn't reproduce this.  Switching to Fancy was slow but it got there.  I tried switching between skins a few times and if flickered the first time (only) but didn't lose the centre.  I wasn't connected to a server.

**Reply by ann0see** [reply]
@DavidSavinkoff every commit after and including f5a58a0 produces this bug for me (as I already mentioned in my initial answer).

It's really strange that this only occurs on my machine.

**Reply by ghost** [reply]
> @DavidSavinkoff every commit after and including [f5a58a0](https://github.com/jamulussoftware/jamulus/commit/f5a58a0d2db7f38d8f6cdc5243b276062c9b2b8b) produces this bug for me (as I already mentioned in my initial answer).
> 
> It's really strange that this only occurs on my machine.

Ok, this is a show-stopper.
[f5a58a0](https://github.com/jamulussoftware/jamulus/commit/f5a58a0d2db7f38d8f6cdc5243b276062c9b2b8b) Has a BUG.
This code ought to be scoured by everyone.
Do I see an extra semicolon in src/audiomixerboard.h ?
If there is one bug, there must be another.
Is there another bug? (I don't know)

**Reply by ann0see** [reply]
Ok. Opened https://github.com/jamulussoftware/jamulus/issues/1238

**Reply by hoffie** [reply]
>  Do I see an extra semicolon in src/audiomixerboard.h ?

Do you mean [line 222](https://github.com/jamulussoftware/jamulus/commit/f5a58a0d2db7f38d8f6cdc5243b276062c9b2b8b#diff-c2dd6e52f5c121333c765285d9a6b9fdf07984b1e14294711d566e6862922792R222)?

That looks correct to me. One semicolon terminates the statement within the function body, the other terminates the declaration. The latter might be optional, but I'm pretty sure it should not cause a change in behaviour.

**Reply by softins** [reply]
I'd like to try this out in a VM today. I haven't looked at the Debian 10 installer yet. Is your system Gnome 2 or Gnome 3? (I don't know if Debian 10 has the choice, but I wanted to be sure in advance).

**Reply by ann0see** [reply]
It's gnome 3: GNOME Shell 3.30.2

**Reply by softins** [reply]
> Maybe I mis-remembered: 
> 
> Now I could reproduce it like that:
> 
> 
> 
> 1. Start Jamulus with fancy mode enabled
> 
> 2. Switch to normal mode while settings is open
> 
> 3. Switch back to fancy mode (maybe you'll need to switch back and forth between the different modes).

I just set up a Debian 10 VM under VirtualBox on Windows. I installed the .deb from the 3.7.0rc1 assets, and set up jackd on a dummy device (unfortunately, the real device was fixed at 44100).

I could reproduce the issue with exactly those steps.

Once the white panel was showing, I could switch to normal or compact mode ok, but switching back to fancy continued to give the white area.

But while the white area was displayed, if I connected to a server, the fancy skin was restored properly as soon as it showed the fader(s). This was true whether the server had recording enabled or not.

Once that had happened, I could disconnect, and then still switch between Normal and Fancy without any problems or white box.

**Reply by softins** [reply]
So, I complile 3.7.0rc1 and it shows the bug repeatably. I compile 3.6.2 and it doesn't. Now using `git bisect` to home in on where it broke.

**Reply by ann0see** [reply]
@softins we've already found the commit. Also see the linked issue.

Good to see that I'm not the only one with this issue

**Reply by softins** [reply]
Ah, ok. I'll corroborate then! ;)

**Reply by dcorson-ticino-com** [reply]
@softins Thanks for the additional information.  We are trying to zero in on the problem here: https://github.com/jamulussoftware/jamulus/issues/1238#issuecomment-798640354
Can you confirm that there are other Linux versions where this is working OK?

**Comment by jujudusud** [comment]
As far as I am able to check, this RC1 is ok on Archlinux with LTS kernel.

**Reply by ann0see** [reply]
Great thank you!

**Comment by jerogee** [comment]
Client on MacOS (Mojave) works well! 👍 

Spotted a bug (#1245) and suggested a tiny translation improvement (#1246).

**Comment by pljones** [comment]
![image](https://user-images.githubusercontent.com/1549463/111080584-03c59100-84f7-11eb-9da9-8140902687b2.png)
Don't know whether there's any point addressing this before release.  It might be nice to move "[ ] Enable Small Network Buffers" over above the network readings ("Audio Stream Rate", etc) or to the top of Misc.

**Reply by softins** [reply]
Actually, I don’t understand why we need the checkbox at all. As far as I have been able to determine, it has no effect when buffers is set to 128 or 256, and if it is not checked, 64 actually behaves the same as 128. So really, it should be implicitly on for 64, and off for 128 or 256. Have I missed anything?

**Reply by dcorson-ticino-com** [reply]
I don't know except for the fact that there are (or maybe were in the mean time) older servers that did not work with small buffer enabled and 64sample, but did work with 64sample and small buffers not enabled.  Because of that I thought the positioning is good.

**Reply by pljones** [reply]
It's needed because it causes problems sometimes.  If you're playing bass, for example, you might need to turn it off.  If you switch over to lead guitar, you could probably benefit from turning it on.  (Yes, it's noticeable, reproducible and Volker was aware of the point when it was introduced.  It's an artefact of passing low pitch tones through the OPUS codec - more cutting causes more "sudden steps" to be possible.)

**Reply by softins** [reply]
What I mean is that it seems turning off SNB and remaining on 64 is identical to just switching to 128.

**Reply by pljones** [reply]
Mmm... I guess so -- 64 LPCM samples arriving and being demanded at twice the rate as 128 OPUS samples departing and arriving.

I think the point is that any other audio buffering in the system is at 64 samples, even if the OPUS/network buffer is 128 samples, if that's your audio buffer size, potentially keeping your overall latency down.

Possibly it could stay in "Soundcard" but move under the buffer size and only be enabled if the buffer size is 64 samples?  (Note, if I'm using ReaRoute with Reaper set to 64 samples, changing buffer size in Jamulus is disabled.)

Anyway, if it needs this much discussion, it's not for 3.7.0.

**Comment by ann0see** [comment]
Locked because this is outdated
